src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8389,40 +8469,179 @@ protected MethodGroupResolution BindExtensionMethod(
+                    // (regardless of whether a best applicable method could be determined) then our search is complete.
+                    // Otherwise, store aside the first non-applicable result and continue searching for an applicable result.
+                    if (extensionResult.HasAnyApplicableMethod)
+                    {
+                        firstResult.Free();
+                        diagnostics.Free();
+                        return extensionResult;
+                    }
+                    else if (firstResult.IsEmpty)
+                    {
+                        firstResult = extensionResult;
+                    }
+                    else
+                    {
+                        // Neither the first result, nor applicable. No need to save result.
+                        extensionResult.Free();
+                    }
+                }
+
+                if (tryResolveExtensionMethod(this, expression, memberName, analyzedArguments, left, typeArgumentsWithAnnotations,
+                    options, returnRefKind, returnType, in callingConvention, diagnostics, scope,
+                    ref actualArguments, out MethodGroupResolution extensionMethodResult))
+                {
+                    // If the search in the current scope resulted in any applicable method (regardless of whether a best
+                    // applicable method could be determined) then our search is complete. Otherwise, store aside the
+                    // first non-applicable result and continue searching for an applicable result.
+                    if (extensionMethodResult.HasAnyApplicableMethod)
+                    {
+                        firstResult.Free();
+                        actualArguments?.Free();
+                        diagnostics.Free();
+                        return extensionMethodResult;
+                    }
+                    else if (firstResult.IsEmpty)
+                    {
+                        firstResult = extensionMethodResult;
+                    }
+                    else
+                    {
+                        // Neither the first result, nor applicable. No need to save result.
+                        extensionMethodResult.Free();
+                    }
+                }
+            }
+
+            Debug.Assert((actualArguments == null) || !firstResult.IsEmpty);
+            actualArguments?.Free();
+            diagnostics.Free();
+
+            return firstResult;
+
+            static bool tryResolveExtensionMember(Binder binder, SyntaxNode expression, string memberName, AnalyzedArguments? analyzedArguments, BoundExpression left,
+                ImmutableArray<TypeWithAnnotations> typeArgumentsWithAnnotations, OverloadResolution.Options options, RefKind returnRefKind, in CallingConventionInfo callingConvention,
+                TypeSymbol? returnType, BindingDiagnosticBag diagnostics, ExtensionScope scope, out MethodGroupResolution extensionResult)
```

Comment:
>BindingDiagnosticBag diagnostics [](http://example.com/codeflow?start=40&length=32)

This approach doesn't feel right. Why would we want to attach an unrelated diagnostics to ```MethodGroupResolution``` that we create. The ```diagnostics``` probably contains errors from previous iterations of the loop above, i.e. a different extension.